concepts/booleans/about.md

```diff
@@ -19,18 +75,43 @@ values into a boolean value.
 
 Unlike some other languages, `0` is not `false` and non-zero is not `true`.
 
-Rust supports three [logical binary operators][logical binary operators]: `&` (Logical And), `|` (Logical Or), `^` (Logical Xor).
+Rust supports three [logical binary operators][logical binary operators]: `&`
+(Logical And), `|` (Logical Or), `^` (Logical Xor).
+
+- `|` (Logical Or)
+    | `a` | `b` | `a \| b` |
+    |- | - | - |
+    | `true` | `true` | `true` |
+    | `true` | `false` | `true` |
+    | `false` | `true` | `true` |
+    | `false` | `false` | `false` |
+
+- `&` (Logical And)
+    | `a` | `b` | [`a & b` |
+    |- | - | - |
+    | `true` | `true` | `true` |
+    | `true` | `false` | `false` |
+    | `false` | `true` | `false` |
+    | `false` | `false` | `false` |
+
+- `^` (Logical Xor)
+    | `a` | `b` | `a ^ b` |
+    |- | - | - |
+    | `true` | `true` | `false` |
+    | `true` | `false` | `true` |
+    | `false` | `true` | `true` |
+    | `false` | `false` | `false` |
 
-Rust supports two [boolean operators][lazy boolean operators]: `||` (Or), `&&` (And). They differ from `|` and `&` in that the right-hand operand
-is only evaluated when the left-hand operand does not already determine the result of the expression. That is, `||` only evaluates
-its right-hand operand when the left-hand operand evaluates to `false`, and `&&` only when it evaluates to `true`.
+The operators `||` (OR), `&&` (AND) differ from `|` and `&`.
```

Comment:
I feel like we should omit any mention of the bitwise operators in the concept about booleans. I've only ever seen bitwise operators used with unsigned integer types, where they operate on bit patterns.

For booleans, as is correctly described here, the difference between `|` and `||` etc. is very small - it's ony about the lazy evaluation. And a program which relies on lazy / eager evaluation of it's boolean expressions because the operands are side-effectful... is a terrible program in my opinion.

In other languages, it may be common to use the lazy boolean operators for control flow, but not in Rust.

With all that in mind, talking about the bitwise operators for booleans at all seems confusing at best, and suggestive of bad practices at worst.

Opinions? @MatthijsBlom

Reply:
I fully agree that some _Operations on Bits_ concept node would be a better place for introducing the bitwise operators.

---

> And a program which relies on lazy / eager evaluation of it's boolean expressions because the operands are side-effectful... is a terrible program in my opinion.

Not exactly a counterexample &ndash; because it does not involve side effects &ndash; but: e.g. `if some_list and predicate(some_list[0])` is a fairly common construct in Python. The `some_list` checks that it is not empty; only if it nonempty is `some_list[0]` looked up, avoiding a runtime error.

More generally, successful evaluation of subsequent propositions might depend on prior propositions holding.